Src/PerformanceCollector/Shared/Implementation/IPerformanceCollector.cs

```diff
@@ -31,5 +30,12 @@ internal interface IPerformanceCollector
         /// <param name="error">Captures the error logged.</param>
         /// <param name="blockCounterWithInstancePlaceHolder">Boolean that controls the registry of the counter based on the availability of instance place holder.</param>
         void RegisterCounter(string perfCounter, string reportAs, bool isCustomCounter, out string error, bool blockCounterWithInstancePlaceHolder);
+
+        /// <summary>
+        /// Removes a counter.
+        /// </summary>
+        /// <param name="perfCounter">Name of the performance counter to remove.</param>
+        /// <param name="reportAs">ReportAs value of the counter to remove.</param>
+        void RemoveCounter(string perfCounter, string reportAs);
```

Comment:
I wish I could. The problem here is that the counters are updated by multiple UX sessions, and we must ensure that other UX sessions (which are most likely viewed by other people) don't experience any sort of interruption when one UX session changes configuration. That's why we can't just recreate everything from scratch here, and must maintain state.
We do do snapshots for the metric configuration since there we can recreate the accumulator exactly the same as before - and it doesn't cost anything. Here, counter creation is very expensive, plus the first read upon creation returns 0.